portfolio/src/main/webapp/script.js

```diff
@@ -147,6 +149,42 @@ function drawChart() {
       });
 }
 
+/**  Hides all forms if user not logged in */
+async function checkLogin() {
+  fetch('/login').then((response) => response.json()).then((loginString) => {
+    const loginEmail = loginString.email;
+    const loginLink = document.getElementById('login-link');
+    loginLink.innerHTML = '';
+
+    if (loginString.loggedIn) {
+      loginLink.appendChild(createLinkElement(
+          loginString.link, `You are logged in as ${loginString.email}. `));
+      loginLink.appendChild(
+          createLinkElement(loginString.link, 'Log out here.'));
+
+      // get comments written by logged in user
+      getComment(loginEmail);
+    } else {
+      loginLink.appendChild(createLinkElement(
+          loginString.link, 'Log in here to interact with the page!'));
+      const commentForm = document.getElementById('commentForm');
+      commentForm.style.display = 'none';
+      const mapForm = document.getElementById('map-form');
+      mapForm.style.display = 'none';
+      const chartForm = document.getElementById('chart-form');
+      chartForm.style.display = 'none';
+    }
+  });
+}
+
+// Creates link element for log in / out
+function createLinkElement(link, text) {
+  const linkElement = document.createElement('a');
```

Comment:
Can you use a meaningful word here instead of 'a'?

Reply:
@liangqigong No, this is a built in HTML tag -- <a href>, using a different word won't work for HTML

Reply:
Sorry I see. Thx.